micromanager_gui/_gui_objects/_property_widget.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, Optional
+
+from pymmcore_plus import PropertyType
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import QComboBox, QHBoxLayout, QLineEdit, QWidget
+from superqt import QLabeledDoubleSlider, QLabeledSlider
+
+from .._core import get_core_singleton
+
+
+class PropertyWidget(QWidget):
+    def __init__(
+        self,
+        device_label: str,
+        prop_name: str,
+        parent: Optional[QWidget] = None,
+        orientation=Qt.Orientation.Horizontal,
+    ) -> None:
+        super().__init__(parent)
+        self._mmc = get_core_singleton()
+
+        if device_label not in self._mmc.getLoadedDevices():
+            raise ValueError(f"Device not loaded: {device_label!r}")
+
+        if not self._mmc.hasProperty(device_label, prop_name):
+            names = self._mmc.getDevicePropertyNames(device_label)
+            raise ValueError(
+                f"Device {device_label!r} has no property {prop_name!r}. "
+                f"Availble property names include: {names}"
+            )
+
+        self._device_label = device_label
+        self._prop_name = prop_name
+        self._prop_type = self._mmc.getPropertyType(device_label, prop_name)
+        if self._prop_type is PropertyType.Float:
+            self._value_widget = QLabeledDoubleSlider(orientation)
+        elif self._prop_type is PropertyType.Integer:
+            self._value_widget = QLabeledSlider(orientation)
+        elif allowed := self._mmc.getAllowedPropertyValues(device_label, prop_name):
+            self._value_widget = QComboBox()
+            self._value_widget.addItems(allowed)
+            self._value_widget.setValue = self._value_widget.setCurrentText
+            self._value_widget.value = self._value_widget.currentText
+        else:
+            self._value_widget = QLineEdit()
+            self._value_widget.setValue = self._value_widget.setText
+            self._value_widget.value = self._value_widget.text
```

Comment:
@tlambert03 should we add something like `self._value_widget.changed =` to each widget? (e.g. if it is a Slider `self._value_widget.changed = self._value_widget.valueChanged`).

This way we can add a new method that is called when the widget wall is changed:
```
def _on_wdg_changed(self, value: Any):
        self._value_widget.setValue(value)
```
and connect it like that:
`self._value_widget.changed.connect(self._on_wdg_changed)`

Reply:
see the update I just pushed... it establishes a `PPropValueWidget` protocol that makes sure they all have the same interface